src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -702,7 +702,7 @@ private BoundExpression MakeObjectInitializerMemberAccess(
                          _compilation.Conversions.HasImplicitConversionToOrImplementsVarianceCompatibleInterface(rewrittenReceiver.Type, memberSymbol.ContainingType, ref discardedUseSiteInfo, out _));
             // It is possible there are use site diagnostics from the above, but none that we need report as we aren't generating code for the conversion
 #endif
-            rewrittenReceiver = _factory.ConvertReceiverForExtensionMemberIfNeeded(memberSymbol, rewrittenReceiver);
+            rewrittenReceiver = this.ConvertReceiverForExtensionMemberIfNeeded(memberSymbol, rewrittenReceiver);
```

Comment:
>this.ConvertReceiverForExtensionMemberIfNeeded(memberSymbol, rewrittenReceiver); [](http://example.com/codeflow?start=32&length=80)

Is there a test that would fail for this code path without the change?

Reply:
I was not able to find one. 
We need:
1. a conversion that is acceptable for the extension parameter: those are identity, boxing, implicit reference, implicit span, or implicit tuple (see `IsValidExtensionMethodThisArgConversion`)
2. the conversion needs to be non-trivial (ie. needs lowering): we're left with implicit span or implicit tuple
3. the receiver type of the property should not be a struct (see logic in `CheckPropertyValueKind` involving `RequiresVariableReceiver`): we're left with no options

The `ExtensionMemberLookup_ObjectInitializer_Conversion_02` test illustrates the ROS scenario. I'll add a couple similar tests